### PR TITLE
Update Libraries

### PR DESCRIPTION
**Pull request by CoolSpy3**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -8,12 +8,13 @@ jobs:
...
       with:
         submodules: 'recursive'
-    - name: Set up JDK 11
-      uses: actions/setup-java@v1
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
       with:
-        java-version: 11
+        distribution: 'temurin'
+        java-version: 17
```
Comment (brettle) [inline .github/workflows/gradle.yml:18]: As far as I can tell, this should be compatible with the Java 11 that 2023 WPILib and 2023RobotCode uses but I'd still recommend waiting until post-season to make this move.

Comment (CoolSpy3) [reply .github/workflows/gradle.yml:18]: [Bundled Java version has been bumped to 17 from 11](https://docs.wpilib.org/en/stable/docs/yearly-overview/yearly-changelog.html)

Comment (brettle) [reply .github/workflows/gradle.yml:18]: Missed that. I just saw that RobotCode2023 build.gradle says "targetCompatibility = JavaVersion.VERSION_11".

Your call. If it were me, I'm honestly not sure what I'd do. I'd still be worried about what might be an unnecessary change at this late date, but I'd also wonder whether using 17 might address some other issue we haven't detected yet.

**Review by brettle — DISMISSED** [review]
Approving but see my comment about the risk of merging. Again... your call.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (CoolSpy3) [reply .github/workflows/gradle.yml:18]: Hmm... Interesting. It appears that the bundled java version is 17, but the project still generates with target version 11. We can hold off on merging until after comp. I mostly made this PR to address some GitHub Actions deprecations.

**Pull request merged**